### ISSUE TITLE
IAR compiled binary fails to run on Dragonfly

### ISSUE THREAD

**Issue by sarahmarshy** [ARMmbed/mbed-os-example-cellular#18]
I am seeing no output when executing an IAR compiled binary. Though, it will run if exported to IAR and flashed through the IDE. The exported project will be compiled with debugging flags, so there might be an issue there. 

For reference, this is my iar version:
```shell
C:\Repos\mbed-os-example-cellular>iccarm --version
IAR ANSI C/C++ Compiler V7.80.1.11864/W32 for ARM
```

**Comment by hasnainvirk** [comment]
@sarahmarshy  There is an issue with MultiTech UDK2.0. Sometimes the binary did not get flashed properly and the software gets stuck in bootloader. Power cycle the device. Plugging-off the power altogather. Reset button wouldn't work at that stage.

**Comment by sarahmarshy** [comment]
@hasnainvirk I have tried that multiple times. I am immediately able to flash and run an ARM compiled binary with the board in the same state. I have also seen the behavior you are talking about, usually it is accompanied by a 'fail.txt' file in the drive as well as the binary sitting in the drive without having been 'consumed', as it were.

**Comment by sarahmarshy** [comment]
This is happening reliably, even after power cycles.

**Comment by hasnainvirk** [comment]
@sarahmarshy Can you please open a Jira Ticket for that ?
I don't have traction for that on Jira. This should be looked into. Thank you for catching it, please elaborate your test setup as well in Jira ticket for reproduciblity.

**Comment by sarahmarshy** [comment]
@hasnainvirk Where am I supposed to create a Jira ticket?

**Comment by hasnainvirk** [comment]
@sarahmarshy Ashok opened a ticket here https://jira.arm.com/browse/ONME-3109

**Comment by ashok-rao** [comment]
Slight correction @hasnainvirk , that ticket was opened by @sarahmarshy only.. :-)

**Comment by hasnainvirk** [comment]
@sarahmarshy @ashok-rao I used mbed-os-5.5.0-rc2-oob branch. And it all worked fine. Can you please elaborate what exactly did you see ? Can you still reproduce it with the above mentioned branch ?

**Comment by hasnainvirk** [comment]
My steps:
i) Set Path for IAR compile in mbed_settings.py
ii) Checkout mbed-os-5.5.0-rc2-oob branch
iii) mbed compile -m MTS_DRAGONFLY_F411RE -t IAR
iv) Copy the binary 
v) Run ..

Trying now the export ..

**Comment by hasnainvirk** [comment]
@ashok-rao @sarahmarshy With a debug build I will get an assertion:
```
mbed-os-example-cellular, Connecting...
Mutex 20019cc0 error -6: Not allowed in ISR context
```
This is a known issue, documented in release notes.  Fix is available here, https://github.com/ARMmbed/mbed-os/pull/4538

**Comment by hasnainvirk** [comment]
@sarahmarshy @ashok-rao  What is the status of this after #4538 ?
If the issue is resolved, please close this issue ?

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTCELL-64

**Comment by MarceloSalazar** [comment]
@sarahmarshy could you please confirm fix with mbed OS 5.6 rc2

**Comment by MarceloSalazar** [comment]
Please disregard. It's confirmed to work in 5.6
